drive.go

```diff
@@ -144,15 +145,15 @@ func (ds *DriveService) ListChildren(ctx context.Context, driveID string, filter
 	return items, resp, nil
 }
 
-// Search the hierarchy of items for items matching a query
-func (ds *DriveService) Search(ctx context.Context, q string, filter string) (*Items, *http.Response, error) {
-	q = url.QueryEscape(q)
-
-	if filter != "" {
-		q = q + "&filter=" + url.QueryEscape(filter)
+// Search the hierarchy of items for all items the user has access to.
+func (ds *DriveService) Search(ctx context.Context, q string) (*Items, *http.Response, error) {
+	// We need to escape single quotes with a double single
+	// quotes, otherwise API will fail with unterminated
+	if strings.Contains(q, "'") {
+		q = strings.ReplaceAll(q, "'", "''")
 	}
 
-	req, err := ds.newRequest("GET", "/drive/root/view.search?q="+q, nil, nil)
+	req, err := ds.newRequest("GET", fmt.Sprintf("/drive/search(q='%s')", url.PathEscape(q)), nil, nil)
```

Comment:
what a weird API URL 🙄

Reply:
> I think this looks okay!
> 
> Do you also need to vendor your dependencies? `go mod vendor`, and commit those?

@ferglor Ah I'm not sure to be honest, should we still be committing the vendor directory with go modules ?

Reply:
I think we only need to vendor for applications, not libraries